lib/tasks/say.js

```diff
@@ -193,8 +193,25 @@ class TaskSay extends TtsTask {
               }).catch((err) => this.logger.info({err}, 'Error adding file to cache'));
             }
 
+            function extractResponseCode(evt) {
+              for (const key in evt) {
+                if (key.startsWith('variable_tts_') && key.endsWith('_response_code')) {
+                  return evt[key];
+                }
+              }
+              return 0;
+            }
+
+            const response_code = extractResponseCode(evt);
```

Comment:
will this work for all vendors?  What about vendors, now or future that don't use http?

Reply:
This will work with all vendors that use http and return variable_tts_<vendor> _response_code.
to support all case, we should keep the legacy code and do trust on variable_tts_error only

inside freeswitch modules, we have to update all vendors that if response code is not 2xx, variable_tts_error should have value so that fallback will happen on feature server.